test/lib/Streamly/Test/Prelude/Common.hs

```diff
@@ -290,6 +291,18 @@ constructWithFromIndicesM op len =
         streamEffect <- run $ readIORef mvl
         listEquals (==) streamEffect list
 
+constructfromWAsync ::
+    S.WAsyncT IO Int -> S.WAsyncT IO Int-> [Int] -> Property
+constructfromWAsync s1 s2 res =
```

Comment:
Can you also check the asyncness of the code?
An `IORef` to capture the effects would work.
```
gen i = threadDelay i >> addToIORefStuff >> return i
```
Then you can use `fromListM` and check the IORef for the expected order of effects.